[x-plane] Build error on x64-windows-static

Package: x-plane:x64-windows-static@4.0.1

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.42.34435.0
-    vcpkg-tool version: 2024-12-09-1005b78fa1bf1dde1a20c2734cba4ea61ca94d9a
    vcpkg-scripts version: 1420fe2 2025-01-09 (5 days ago)

**To Reproduce**

`vcpkg install `

**Failure logs**

```
Downloading XPSDK401.zip
Failed to download XPSDK401.zip.
If you are using a proxy, please ensure your proxy settings are correct.
Possible causes are:
1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable to `https//address:port`.
This is not correct, because `https://` prefix claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr, etc...) is an HTTP proxy.
Try setting `http://address:port` to both HTTP_PROXY and HTTPS_PROXY instead.
2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings set by your proxy software. See, https://github.com/microsoft/vcpkg-tool/pull/77
The value set by your proxy might be wrong, or have same `https://` prefix issue.
3. Your proxy's remote server is our of service.
If you've tried directly download the link, and believe this is not a temporay download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
to report this upstream download server failure.
error: https://developer.x-plane.com/wp-content/plugins/code-sample-generation/sample_templates/XPSDK401.zip: failed: status code 404
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:231 (message):
  Download failed, halting portfile.
Call Stack (most recent call first):
  C:/Users/jamarr/AppData/Local/vcpkg/registries/git-trees/be6ff276d2a18e64a2cf91a50ef1a53d26a33fa8/portfile.cmake:4 (vcpkg_download_distfile)
  scripts/ports.cmake:196 (include)



```

**Additional context**

<details><summary>vcpkg.json</summary>

```
{
  "dependencies": [
        "libpng",
        "freetype",
        "pcre",
        "x-plane"
      ]
}
```
</details>

**Other information that may be useful (release notes, etc...)**
Laminar, the publisher, has removed their download link for v4.0.1. They have gone through a few v4.0.x revisions, and eventually landed on [v4.1.1](https://developer.x-plane.com/wp-content/plugins/code-sample-generation/sdk_zip_files/XPSDK411.zip). Some previous versions are available from their [X-Plane SDK Downloads](https://developer.x-plane.com/sdk/plugin-sdk-downloads/) page. The fact that they delete older, possibly defective, SDK versions seems problematic from a package-manager context, so they will likely do this again. 

It seems that v4.1.0 has been out for at least 6mo and v4.1.1 for at least 3mo, so the latest versions _appear_ to be stable. The recipe for this project should attempt to download the latest available version it is aware of. This means that if v4.1.1 is removed, it would fall back to v4.1.0; if that is removed, it would fall back to v4.0.0. I recognize this may result in some newer SDK features becoming unavailable. Still, many SDK users are not necessarily using cutting-edge features anyway, and having an SDK version to build against is better than having no SDK at all.

--- a/ports/x-plane/portfile.cmake
+++ b/ports/x-plane/portfile.cmake
@@ -3,7 +3,7 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 set(XPSDK_VERSION "401")
 vcpkg_download_distfile(
     XPLANE_SDK_ZIP
-    URLS "https://developer.x-plane.com/wp-content/plugins/code-sample-generation/sample_templates/XPSDK${XPSDK_VERSION}.zip"
+    URLS "https://developer.x-plane.com/wp-content/plugins/code-sample-generation/sdk_zip_files/XPSDK${XPSDK_VERSION}.zip"
     FILENAME "XPSDK${XPSDK_VERSION}.zip"
     SHA512 8e00789befd15f5b1cb4f426ddf9c3f7f021c5fba50b907e8af5fbf09abbc362804b5d1543332855d01e8ae91b9c50a55933e63df6e11e88e58c10ca8f949bf4
 )

--- a/ports/x-plane/vcpkg.json
+++ b/ports/x-plane/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "x-plane",
   "version": "4.0.1",
+  "port-version": 1,
   "description": "The X-Plane Plugin SDK",
   "homepage": "https://developer.x-plane.com/sdk/",
   "license": null,

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -9770,7 +9770,7 @@
     },
     "x-plane": {
       "baseline": "4.0.1",
-      "port-version": 0
+      "port-version": 1
     },
     "x264": {
       "baseline": "0.164.3108",

--- a/versions/x-/x-plane.json
+++ b/versions/x-/x-plane.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "605d085f1c9ef0cf423d20a5dc574003b42a4e2e",
+      "version": "4.0.1",
+      "port-version": 1
+    },
+    {
       "git-tree": "be6ff276d2a18e64a2cf91a50ef1a53d26a33fa8",
       "version": "4.0.1",
       "port-version": 0

